src/core/coap/coap.cpp

```diff
@@ -662,6 +667,15 @@ void Coap::ProcessReceivedRequest(Header &aHeader, Message &aMessage, const Ip6:
     }
 
 exit:
+
+    if (error != kThreadError_None)
+    {
+        if (response != NULL)
```

Comment:
style: avoid additional nesting by putting both conditions in same if-statement.